hazelcast/src/main/java/com/hazelcast/client/impl/connection/AddressProvider.java

```diff
@@ -29,9 +30,15 @@ public interface AddressProvider {
     /**
      * @return The possible member addresses to connect to.
      * @throws Exception when a remote service can not provide addressee
+     * @deprecated use {@link #loadAddresses(ClientConnectionProcessListener)} instead
      */
+    @Deprecated
```

Comment:
This is an internal API. If we were to only have the other method, instead of deprecating it, we should remove it IMHO

Reply:
removed the old signature, thanks